obspy/clients/httpproxy.py

```diff
@@ -0,0 +1,124 @@
+
+
+def get_proxy_tuple():
+    """
+    Return system http proxy as a urlparse.urlparse tuple or () if unset.
+    """
+    proxydict = urllib.getproxies()
+    proxystr = proxydict.get('http') or proxydict.get('https') or ''
+    if proxystr:
+        proxy = urlparse.urlparse(proxystr)
+    else:
+        proxy = ()
+
+    return proxy
+
+
+def valid_address(addr):
+    """ Verify that an IP/port tuple is valid """
+    is_valid = (isinstance(addr, (list, tuple)) and
+                len(addr) == 2 and
+                isinstance(addr[0], (str, native_str)) and
+                isinstance(addr[1], int))
+    return is_valid
+
+
+def http_proxy_connect(address, proxy, auth=None, headers={}):
+    """
+    Establish a socket connection through an HTTP proxy.
+
+    Arguments:
+      address (required)     = The address of the target
+      proxy (def: None)      = The address of the proxy server
+      auth (def: None)       = A tuple of the username and password used for
+                               authentication
+      headers (def: {})      = A set of headers that will be sent to the proxy
+
+    Returns:
+      A 3-tuple of the format:
+        (socket, status_code, headers)
+      Where `socket' is the socket object, `status_code` is the HTTP status
+      code that the server returned and `headers` is a dict of headers that the
+      server returned.
+
+    """
+    if not valid_address(address):
+        raise ValueError('Invalid target address')
+
+    if not valid_address(proxy):
+        raise ValueError('Invalid proxy address')
+
+    headers = {'host': address[0]}
+
+    if auth is not None:
+        if isinstance(auth, str):
+            headers['proxy-authorization'] = auth
+        elif auth and isinstance(auth, (tuple, list)) and len(auth) == 2:
+            proxy_authorization = 'Basic ' + b64encode('%s:%s' % auth)
+            headers['proxy-authorization'] = proxy_authorization
+        else:
+            raise ValueError('Invalid authentication specification')
+
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.connect(proxy)
+    fp = s.makefile('r+')
+
+    fp.write('CONNECT %s:%d HTTP/1.0\r\n' % address)
+    fp.write('\r\n'.join('%s: %s' % (k, v) for (k, v) in headers.items()) +
+             '\r\n\r\n')
+    fp.flush()
+
+    statusline = fp.readline().rstrip('\r\n')
+
+    if statusline.count(' ') < 2:
+        fp.close()
+        s.close()
+        raise IOError('Bad response')
+    version, status, statusmsg = statusline.split(' ', 2)
+    if version not in ('HTTP/1.0', 'HTTP/1.1'):
+        fp.close()
+        s.close()
+        raise IOError('Unsupported HTTP version')
+    try:
+        status = int(status)
+    except ValueError:
+        fp.close()
+        s.close()
+        raise IOError('Bad response')
+
+    response_headers = {}
+
+    while True:
+        l = fp.readline().rstrip('\r\n')
+        if l == '':
+            break
+        if ':' not in l:
+            continue
```

Comment:
Looks like it just wants to try the next line, not to exit the loop.  Again, I'm not familiar with sockets.